js/script-loader.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+// loadScript(script: Array<Promise|string>|string): Promise
+// eslint-disable-next-line no-var
+var loadScript = (function () {
+  const cache = new Map();
+
+  function inject(file) {
+    if (!cache.has(file)) {
+      cache.set(file, doInject(file));
```

Comment:
I haven't used it alot, but wouldn't it be better make the cache into a `Set()`, then you wouldn't need to check for duplicates.

Reply:
I use `Map` here to save the downloading process. For example:
```js
const pending1 = loadScript('test.js');
const pending2 = loadScript('test.js');
```
`doInject` would be only called once so `pending2` won't create another request and both `pending1` and `pending2` will be resolved at the same time.

Reply:
Er, why do we need another script loader in addition to injectScript and onDOMscripted?

Reply:
He renamed `onDOMscripted` and moved it into a separate file.